### ISSUE TITLE
Does it make sense to have an option to clean PV before discovering it as a new PV?

### ISSUE THREAD

**Issue by cofyc** [kubernetes-sigs/sig-storage-local-static-provisioner#143]
Current lifecycle:

```
- raw disks -> mount points -> PV (available) -> PV (bound) -> PV (released)
                               ^                                     |   
                               | <---------- clean <-----------------|                                         
```

New lifecycle (optional):

```
- raw disks -> mount points -> [clean] -> PV (available) -> PV (bound) -> PV (released)
                                          ^                                     |   
                                          | <---------- clean <-----------------|
```

**Comment by cofyc** [comment]
use cases:

- if we clean the volume before provisioning it as a new PV into Kubernetes, it will be safe to delete the PV objects and don't worry about the data will be leaked. (https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/126#issuecomment-511651492)

**Comment by yuyulei** [comment]
another case:

- you use cloud storage as local pv, after pvc&pod deleted, and cleanup pv successfully. But actually, cloud storage not works as expected. Then provisioner creates the pv with same name and path. Later another pod uses the pv, it finds some data not belong to itself. 

so it really make sense, at least, check there are some data before pv available.

**Comment by msau42** [comment]
Yes I think this is a good idea, although this will cause the recycling flow to double for very large disks. We can add it as a new option

/help
/kind feature

**Comment by Bessonov** [comment]
@vamsikrishna-siddu I'm not using storage-local-static-provisioner at the moment, but I think it would make sense.
